Jenkinsfile.update

```diff
@@ -0,0 +1,211 @@
+    }
+    parameters {
+        // GIT-INFORMATION
+        string(name: 'BUILD_BRANCH_NAME', defaultValue: 'master', description: 'Which branch to build? Set if you are not on a multibranch pipeline.')
+        string(name: 'GIT_AUTHOR', defaultValue: 'kiegroup', description: 'Which Git author repository ?')
+
+        // RELEASE INFORMATION
+        booleanParam(name: 'RELEASE', defaultValue: true, description: 'Is this build for a release?')
+
+        // VERSION UPDATE INFORMATION
+        string(name: 'NEW_SNAPSHOT_VERSION', defaultValue: '', description: 'Setup a specific build display name')
+        string(name: 'MAVEN_DEPENDENCIES_REPOSITORY', defaultValue: '', description: 'Maven repository where to find dependencies if those are not in the default JBoss repository.')
+        string(name: 'MAVEN_SETTINGS_CONFIG_FILE_ID', defaultValue: 'kogito_release_settings', description: 'Maven settings configfile to use in pipeline for Maven commands')
+        string(name: 'KOGITO_VERSION', defaultValue: '', description: 'Optional if not RELEASE. If RELEASE, cannot be empty.')
+
+        // PULLREQUEST INFORMATION - Bot author information. Set as params for easy testing.
+        string(name: 'GIT_AUTHOR_BOT', defaultValue: 'bsig-gh-bot', description: 'From which author should the PR be created ?')
+        string(name: 'BOT_CREDENTIALS_ID', defaultValue: 'bsig-gh-bot', description: 'Credentials for PR creation')
+    }
+
+    stages {
+        stage('Clean workspace') {
+            steps {
+                cleanWs()
+            }
+        }
+        stage('Checkout repositories') {
+            steps {
+                checkoutRepo(optaplannerRepository, getBuildBranch())
+                checkoutRepo(vehicleRoutingRepository, getBuildBranch())
+                checkoutRepo(employeeRosteringRepository, getBuildBranch())
+                checkoutRepo(kogitoExamplesRepository, getBuildBranch())
+                checkoutRepo(kogitoAppsRepository, getBuildBranch())
+            }
+        }
+        stage('Prepare for PR') {
+            when {
+                expression { return isRelease() }
+            }
+            steps {
+                prepareForPR(optaplannerRepository)
+                prepareForPR(vehicleRoutingRepository)
+                prepareForPR(employeeRosteringRepository)
+                prepareForPR(kogitoExamplesRepository)
+                prepareForPR(kogitoAppsRepository)
+            }
+        }
+        stage('Update to next SNAPSHOT version') {
+            when {
+                expression { return getProjectVersion() != '' }
+            }
+            steps {
+                script {
+                    maven.mvnVersionsSet(getOptaplannerMavenCommand(), getProjectVersion(), true)
+                    maven.mvnSetVersionProperty(getOptaplannerMavenCommand(), 'version.org.kie.kogito', getKogitoVersion())
+
+                    mavenCleanInstallOptaPlannerParents()
```

Comment:
The new optaplanner snapshot should be also deployed to JBoss Nexus; otherwise, the CI on PRs to the projects depending on optaplanner will fail.